dialer/dialer.go

```diff
@@ -42,7 +42,7 @@ func Dialer(address string, timeout time.Duration) (net.Conn, error) {
 		close(stopC)
 		go func() {
 			dr := <-synC
-			if dr != nil {
+			if dr != nil && dr.c != nil {
```

Comment:
seems reasonable; given this means that `dialer()` returns an error I wonder if there is any reason to log this error, which is effectively coming from `net.DialTimeout()`? But given a timeout error is already being returned and this is in a goroutine it could be messy to try and log.

Reply:
Given we reached the timeout, I don't think it's useful to log any error the dialer might return.